p2p/nat/nat.go

```diff
@@ -91,20 +91,22 @@ func Parse(spec string) (Interface, error) {
 }
 
 const (
-	mapTimeout = 10 * time.Minute
+	DefaultMapTimeout = 10 * time.Minute
 )
 
 // Map adds a port mapping on m and keeps it alive until c is closed.
 // This function is typically invoked in its own goroutine.
+//
+// This does not support using alternate ports.
```

Comment:
It makes sense to me. I think we should define it as an error if the port number passed in as an argument is different from the actual mapped port number.

Reply:
I think it's fine not to worry about this. The reason why it's "not supported" here is because the caller of `Map` has no way of learning about the alternative port assigned by the router. We don't even use `Map` anymore after this PR, so it doesn't really matter.